src/software/jetson_nano/thunderloop.cpp

```diff
@@ -157,34 +160,65 @@ void Thunderloop::runLoop()
 
                 // Handle emergency stop override
                 struct timespec result;
-                ScopedTimespecTimer::timespecDiff(&poll_time,
+
+                clock_gettime(CLOCK_MONOTONIC, &current_time);
+                ScopedTimespecTimer::timespecDiff(&current_time,
                                                   &last_primitive_received_time, &result);
 
                 auto nanoseconds_elapsed_since_last_primitive =
                     result.tv_sec * static_cast<int>(NANOSECONDS_PER_SECOND) +
                     result.tv_nsec;
 
-                // If we haven't received a primitive in a while, override the
-                // current_primitive_ with an estop primitive.
                 if (nanoseconds_elapsed_since_last_primitive >
                     static_cast<long>(PRIMITIVE_MANAGER_TIMEOUT_NS))
                 {
-                    primitive_.Clear();
-                    *(primitive_.mutable_estop()) = emergency_stop_override;
+                    primitive_executor_.clearCurrentPrimitive();
                 }
```

Comment:
Could you add a todo for #2726 either in this file or in primitive executor? 
We should also not move if a World hasn't been received in a while.